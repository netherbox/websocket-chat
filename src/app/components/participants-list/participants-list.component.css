ul {
  list-style: none;
  padding: 0;
  margin: 0;
}

li {
  padding: 20px;
  font-weight: 500;
  border-bottom: 1px solid #dcdfe2;
}
