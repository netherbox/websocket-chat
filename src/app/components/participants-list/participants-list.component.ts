import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'app-participants-list',
  templateUrl: './participants-list.component.html',
  styleUrls: ['./participants-list.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class ParticipantsListComponent {
  @Input()
  public participants: Array<string> = [];
}
