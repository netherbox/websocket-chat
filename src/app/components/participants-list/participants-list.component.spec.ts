import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ParticipantsListComponent } from './participants-list.component';

describe('ParticipantsListComponent', () => {
  let component: ParticipantsListComponent;
  let fixture: ComponentFixture<ParticipantsListComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ParticipantsListComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(ParticipantsListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should successfully create component with empty list', () => {
    expect(component).toBeTruthy();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  describe('participants', () => {
    it('should render participants list', () => {
      component.participants = ['test-participants-1', 'test-participants-2'];
      fixture.detectChanges();

      expect(fixture.nativeElement.querySelectorAll('li').length).toEqual(2);
      expect(fixture.nativeElement).toMatchSnapshot();
    });
  });
});
