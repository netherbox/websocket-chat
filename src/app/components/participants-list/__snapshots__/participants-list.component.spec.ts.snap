// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantsListComponent participants should render participants list 1`] = `
<div
  id="root1"
>
  <ul>
    
    <li>
      test-participants-1
    </li>
    <li>
      test-participants-2
    </li>
  </ul>
</div>
`;

exports[`ParticipantsListComponent should successfully create component with empty list 1`] = `
<div
  id="root0"
>
  <ul>
    
  </ul>
</div>
`;
